lib/chef/resource/aws_instance.rb

```diff
@@ -1,10 +1,14 @@
 require 'chef/provisioning/aws_driver/aws_resource_with_entry'
 
 class Chef::Resource::AwsInstance < Chef::Provisioning::AWSDriver::AWSResourceWithEntry
+  # TODO when I try to set this as `Aws` instead of `AWS` I start getting lots of load errors
```

Comment:
This type is used for creating the instance. I would be very wary of using v1 instance in one place and v2 in others; I think you need to sort this out before checking in. I'm happy to pair.